src/code/audio/code_8019AF00.c

```diff
@@ -1953,17 +1957,152 @@ const char sAudioOcarinaUnusedText7[] = "check is over!!! %d %d %d\n";
+        } else {
+            j = Audio_NextRandom() % 9;
+            ocarinaNote = sOoTOcarinaSongNotes[j];
+
+            for (; ocarinaNote[i].length != 0; i++) {
+                sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][i] = ocarinaNote[i];
+            }
+
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][i].pitch = OCARINA_PITCH_NONE;
+            sOcarinaSongNotes[OCARINA_SONG_TERMINA_WALL][i].length = 0;
+            AudioOcarina_MapSongFromNotesToButtons(OCARINA_SONG_TERMINA_WALL, OCARINA_SONG_TERMINA_WALL,
+                                                   sOoTOcarinaSongsNumNotes[j]);
+        }
+    } while (AudioOcarina_TerminaWallValidateNotes(OCARINA_SONG_TERMINA_WALL, OCARINA_SONG_TERMINA_WALL) != 0);
+}
+
+/**
+ * Unused remnant of OoT
+ */
+void AudioOcarina_MemoryGameSetNumNotes(u8 minigameRound) {
+    u8 i;
+
+    if (minigameRound > 2) {
+        minigameRound = 2;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D26C.s")
+    sOcaMemoryGameAppendPos = 0;
+    sOcaMemoryGameEndPos = sOcaMemoryGameNumNotes[minigameRound];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D488.s")
+    for (i = 0; i < 3; i++) {
+        AudioOcarina_MemoryGameGenerateNotes();
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_8019D4F8.s")
+/**
+ * Unused remnant of OoT, Id 14 now represent Goron Lullaby Intro instead of the OoT ocarina memory game
+ */
+#define OCARINA_SONG_MEMORYGAME OCARINA_SONG_GORON_LULLABY_INTRO
+s32 AudioOcarina_MemoryGameGenerateNotes(void) {
+    u32 rndButtonIdx;
+    u8 rndPitch;
+
+    if (sOcaMemoryGameAppendPos == sOcaMemoryGameEndPos) {
+        return true;
+    }
+
+    rndButtonIdx = Audio_NextRandom();
+    rndPitch = sButtonToPitchMap[rndButtonIdx % 5];
+
+    if (sOcarinaSongNotes[OCARINA_SONG_GORON_LULLABY_INTRO][sOcaMemoryGameAppendPos - 1].pitch == rndPitch) {
+        rndPitch = sButtonToPitchMap[(rndButtonIdx + 1) % 5];
+    }
+
+    sOcarinaSongNotes[OCARINA_SONG_GORON_LULLABY_INTRO][sOcaMemoryGameAppendPos].pitch = rndPitch;
+    sOcarinaSongNotes[OCARINA_SONG_GORON_LULLABY_INTRO][sOcaMemoryGameAppendPos].length = 45;
+    sOcarinaSongNotes[OCARINA_SONG_GORON_LULLABY_INTRO][sOcaMemoryGameAppendPos].volume = 0x50;
+    sOcarinaSongNotes[OCARINA_SONG_GORON_LULLABY_INTRO][sOcaMemoryGameAppendPos].vibrato = 0;
+    sOcarinaSongNotes[OCARINA_SONG_GORON_LULLABY_INTRO][sOcaMemoryGameAppendPos].bend = 0;
+
+    sOcaMemoryGameAppendPos++;
+
+    sOcarinaSongNotes[OCARINA_SONG_GORON_LULLABY_INTRO][sOcaMemoryGameAppendPos].pitch = OCARINA_PITCH_NONE;
+    sOcarinaSongNotes[OCARINA_SONG_GORON_LULLABY_INTRO][sOcaMemoryGameAppendPos].length = 0;
+    sOcarinaSongNotes[OCARINA_SONG_GORON_LULLABY_INTRO][sOcaMemoryGameAppendPos + 1].pitch = OCARINA_PITCH_NONE;
+    sOcarinaSongNotes[OCARINA_SONG_GORON_LULLABY_INTRO][sOcaMemoryGameAppendPos + 1].length = 0;
+    if (1) {}
```

Comment:
Did you forgot to use the macro? lol

```suggestion
    if (sOcarinaSongNotes[OCARINA_SONG_MEMORYGAME][sOcaMemoryGameAppendPos - 1].pitch == rndPitch) {
        rndPitch = sButtonToPitchMap[(rndButtonIdx + 1) % 5];
    }

    sOcarinaSongNotes[OCARINA_SONG_MEMORYGAME][sOcaMemoryGameAppendPos].pitch = rndPitch;
    sOcarinaSongNotes[OCARINA_SONG_MEMORYGAME][sOcaMemoryGameAppendPos].length = 45;
    sOcarinaSongNotes[OCARINA_SONG_MEMORYGAME][sOcaMemoryGameAppendPos].volume = 0x50;
    sOcarinaSongNotes[OCARINA_SONG_MEMORYGAME][sOcaMemoryGameAppendPos].vibrato = 0;
    sOcarinaSongNotes[OCARINA_SONG_MEMORYGAME][sOcaMemoryGameAppendPos].bend = 0;

    sOcaMemoryGameAppendPos++;

    sOcarinaSongNotes[OCARINA_SONG_MEMORYGAME][sOcaMemoryGameAppendPos].pitch = OCARINA_PITCH_NONE;
    sOcarinaSongNotes[OCARINA_SONG_MEMORYGAME][sOcaMemoryGameAppendPos].length = 0;
    sOcarinaSongNotes[OCARINA_SONG_MEMORYGAME][sOcaMemoryGameAppendPos + 1].pitch = OCARINA_PITCH_NONE;
    sOcarinaSongNotes[OCARINA_SONG_MEMORYGAME][sOcaMemoryGameAppendPos + 1].length = 0;
    if (1) {}
```

Reply:
I wasn't entirely sure what to do with this. Part of me wanted to leave in `OCARINA_SONG_GORON_LULLABY_INTRO` to show that this code was impacting that song, but then part of me wanted to use the define `OCARINA_SONG_MEMORYGAME` so that it matched OoT's code. So do you think this is better than using Goron Lullaby Intro?